django/utils/autoreload.py

```diff
@@ -285,7 +285,7 @@ def __init__(self):
     def watch_dir(self, path, glob):
         path = Path(path)
         try:
-            path = path.absolute()
+            path = path.resolve(strict=True)
```

Comment:
In the past we removed `strict=True` from `Path.resolve()` calls as it caused permission errors when user didn't have permissions to all intermediate directories, see e39e727ded673e74016b5d3658d23cbe20234d11.

Reply:
We added catching `FileNotFoundError` to avoid crashes related to an issue in Python itself. If this is already fixed in Python (I'd love to bisect an actual change) we should be able to revert fc75694257b5bceab82713f84fe5a1b23d641c3f instead.

Reply:
What about an initial check if the path exists?

```
path = Path(path)
if path.exists():
    path = path.resolve()  # or path.absolute()
else:
    logger.debug(...)
```

This is more what I had in mind when I submitted the bug report.

Reply:
I tried a quick bisect before submitted the PR but came up empty. I’ll see if I can take another pass at it.

Reply:
It was not crucial for this method to check if the path exists. We added `try ... except` to avoid crashes related with an issue in Python run on some Docker images (check out ticket-30647), IMO it's still not fixed. `watch_dir()` is used in `watch_for_translation_changes()` to watch all **potential** directories, so actually checking `exists()` may not be desirable.